main.nf

```diff
@@ -0,0 +1,80 @@
+    include {MULTIQC as MultiQc} from "./modules/nf-core/multiqc/main.nf"
+    include {CUTADAPT as Cutadapt_1} from "./modules/nf-core/cutadapt/main.nf"
+    include {CUTADAPT as Cutadapt_2} from "./modules/nf-core/cutadapt/main.nf"
+    include {MINIMAP2_ALIGN as MiniMap2} from "./modules/nf-core/minimap2/align/main.nf"
+    include {DEEPVARIANT as DeepVariant} from "./modules/nf-core/deepvariant/main.nf"
+    include {WHATSHAP as WhatsHap} from "./modules/local/whatshap.nf"
+    include {SAMTOOLS_INDEX as Samtools_Index} from "./modules/nf-core/samtools/index/main.nf"
+    include {GUNZIP as Gunzip_cutadapt_1} from "./modules/nf-core/gunzip/main.nf" 
+    include {GUNZIP as Gunzip_cutadapt_2} from "./modules/nf-core/gunzip/main.nf"
+    include {GUNZIP as Gunzip_deepvariant} from "./modules/nf-core/gunzip/main.nf"
+    include {BEDTOOLS_BAMTOBED as Bedtools_minimap2} from "./modules/nf-core/bedtools/bamtobed/main.nf"
+    include {CSV_READ as Csv_Read} from "./lib/csv_reader.nf"
+    include {COMBINE_CUTADAPT as Cutadapt} from "./modules/local/combine_cutadapt.nf"
+    include {LINKED_ADAPERS as Linked_Adapters} from "./modules/local/linked_adapters.nf"
+  
+
+//"input output defining"------------------------------------------------------------------------------------------ 
+
+
+    //---------------------------------CSV
+
+    //CSV_package
+    @Grab( 'com.xlson.groovycsv:groovycsv:1.0' )
+    import com.xlson.groovycsv.CsvParser
+
+    //load CSV
+    read_list = Csv_Read(CsvParser.parseCsv(file("${params.csv_file}", checkIfExists: true).text))
+
+
+    //---------------------------------REFERENCE
+    referenceFasta = [[id:"fasta"],file("${params.referenceFasta}", checkIfExists: true)]
+    referenceFastaFai = [[id:"fastafai"],file("${params.referenceFastaFai}", checkIfExists: true)]
+
+
+workflow {
+
+    //trimming adapters first run
+    Cutadapt_1(Channel.fromList(read_list))
+
+    //trimming adapters seconnd run
+    Cutadapt_2(Channel.fromList(read_list))
+
+    //unzipping first trimming
+    Gunzip_cutadapt_1(Cutadapt_1.out.reads)
+
+    //unzipping first trimming
+    Gunzip_cutadapt_2(Cutadapt_2.out.reads)
+
+    //combining cutadapt output
+    Cutadapt(Gunzip_cutadapt_1.out.gunzip.join(Gunzip_cutadapt_2.out.gunzip))
+
+    //quality control with trimmed reads
+    FastQC(Cutadapt.out.combined_fastq)
+
+    //run the alignment of the reads with minimap2
+    MiniMap2(Cutadapt.out.combined_fastq, referenceFasta, true, false, false)
+
+    //index the aligned reads output from minimap2 for deepvariant
+    Samtools_Index(MiniMap2.out.bam)
+
+    //making a bed file to run deepvariant
+    Bedtools_minimap2(MiniMap2.out.bam)
+
+    //run the variantcalling with deepvariant
+    DeepVariant(MiniMap2.out.bam.join(Samtools_Index.out.bai).join(Bedtools_minimap2.out.bed), referenceFasta, referenceFastaFai, [[],[]])
+    
+    //decompress the output vcf file of deepvariant
+    Gunzip_deepvariant(DeepVariant.out.vcf)
```

Comment:
Unzipping should not be necessary. Most tools support bgzip format input. I am quite sure WhatsHap supports it!